### PR TITLE
Optimizing test script get_missed_upgrade_jobs.py

### PR DESCRIPTION
**Pull request by JianLi-RH**
/cc @shellyyang1989 @jianlinliu @liangxia

### CODE REVIEW

File: ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py [hunk 1]
```diff
@@ -67,8 +67,11 @@ def get_missed_profiles(e2e_jobs, upgrade_jobs, mode='y'):
         'hypershift',
         'longduration',
         'mco',
+        'migrate',
+        'migration',
         'netobserv',
         'nokubeadmin',
+        'obo',
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py:74]: aro/mco/obo, they are too short. We don't have profiles with words that has those abbr as part of it, but there is no guarantee we will not in the future.
Perhaps try to check with `-`, e.g, `re.search('-obo', job)` and `re.search('obo-', job)`

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py:74]: Not a blocker. We can do that in this PR, or in a separate PR.

Comment (JianLi-RH) [reply ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py:74]: we have:
`aws-ipi-ovn-ipsec-f2-obo`
I will add `-` before them and I will also add some other improvements, so hold on.
Thanks.

**Comment by JianLi-RH** [comment]
hi @jianlinliu @liangxia @shellyyang1989 I did some improvements on the script, please take a look, thanks. The changes:
1. adding `-` before some keywords, it can fix the issue like we skipped `ibmcloud` unexpectedly 
2. add a new method `addtional_check` which can:

- skip jobs for multi arch which has `disc` and `mixarch`
- skip jobs which has FEATURE_SET = CustomNoUpgrade

For `sdn`, I only skip it for ocp 4.16 and later. @jianlinliu please help confirm.

**Comment by JianLi-RH** [comment]
Here are the test on my local machine:
```
[jianl@jianl-thinkpadt14gen4 release]$ e2e=/home/jianl/1_code/release/ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml
[jianl@jianl-thinkpadt14gen4 release]$ upgrade=/home/jianl/1_code/release/ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__multi-nightly-4.17-upgrade-from-stable-4.16.yaml
```
Test the script `with`  new method:
```
[jianl@jianl-thinkpadt14gen4 release]$ python /home/jianl/1_code/release/ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py $e2e $upgrade
- as: aws-ipi-byo-iam-role-arm-f2
- as: aws-ipi-byo-kms-etcd-encryption-arm-f2
- as: aws-ipi-usertags-custom-sg-arm-f5
- as: azure-ipi-arm-mixarch-f5
- as: azure-ipi-des-day2-arm-f28
- as: azure-ipi-mto-arm-mixarch-f7
- as: gcp-ipi-compact-filestore-csi-capi-arm-f14
- as: gcp-ipi-ovn-ipsec-capi-arm-mixarch-f14
[jianl@jianl-thinkpadt14gen4 release]$
```

Test the script `without` the new method:
```
[jianl@jianl-thinkpadt14gen4 release]$ python /home/jianl/1_code/release/ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py $e2e $upgrade
- as: aws-ipi-byo-iam-role-arm-f2
- as: aws-ipi-byo-kms-etcd-encryption-arm-f2
- as: aws-ipi-disc-priv-arm-mixarch-f2
- as: aws-ipi-usertags-custom-sg-arm-f5
- as: azure-ipi-arm-mixarch-f5
- as: azure-ipi-des-day2-arm-f28
- as: azure-ipi-disc-fullypriv-arm-mixarch-f2
- as: azure-ipi-mto-arm-mixarch-f7
- as: gcp-ipi-basecap-none-additionalcaps-capi-arm-f7
- as: gcp-ipi-compact-filestore-csi-capi-arm-f14
- as: gcp-ipi-disc-priv-oidc-arm-mixarch-f14
- as: gcp-ipi-disk-encryption-capi-arm-f14
- as: gcp-ipi-ovn-ipsec-capi-arm-mixarch-f14
[jianl@jianl-thinkpadt14gen4 release]$ 
```

**Comment by shellyyang1989** [comment]
lgtm

**Comment by shellyyang1989** [comment]
Just recall that Jia added ota dedicated jobs recently, we might need to skip them when adding upgrade jobs.

**Comment by JianLi-RH** [comment]
ok, the new keyword `-ota` will be added in future changes, thanks

**Pull request merged**